test/EDSC/builder-api-test.cpp

```diff
@@ -746,6 +747,39 @@ TEST_FUNC(empty_map_load_store) {
   f.erase();
 }
 
+// CHECK-LABEL: func @affine_if_op
+// CHECK:       affine.if ([[d0:.*]], [[d1:.*]]){{\[}}[[s0:.*]], [[s1:.*]]{{\]}}
+// CHECK-NEXT:  }
+// CHECK:       affine.if ([[d0:.*]], [[d1:.*]]){{\[}}[[s0:.*]], [[s1:.*]]{{\]}}
+// CHECK-NEXT:  } else {
+TEST_FUNC(affine_if_op) {
+  using namespace edsc;
+  using namespace edsc::intrinsics;
+  using namespace edsc::op;
+  auto f32Type = FloatType::getF32(&globalContext());
+  auto memrefType = MemRefType::get({-1, -1}, f32Type, {}, 0);
+  auto f = makeFunction("affine_if_op", {}, {memrefType});
+
+  OpBuilder builder(f.getBody());
+  ScopedContext scope(builder, f.getLoc());
+  // clang-format off
+  ValueHandle zero = constant_index(0), ten = constant_index(10);
+  
+  SmallVector<bool, 4> isEq = {false, false};
+  SmallVector<AffineExpr, 4> affineExprs = {
+    builder.getAffineDimExpr(0) + builder.getAffineSymbolExpr(0), // d0 + s0 >= 0
+    builder.getAffineDimExpr(1) - builder.getAffineSymbolExpr(1), // d1 - s1 >= 0
```

Comment:
Do we need a complicated example here? Can it be a trivial set?

Reply:
Just a case where we use both dims and symbols and passing them to the op args. I will eliminate the arithmetic ops in the inequalities (e.g. `d0 >=0`, `s0 >=0` ... )

Reply:
Done.